src/Physics/XSectionIntegration/GSLXSecFunc.cxx

```diff
@@ -187,6 +187,46 @@ ROOT::Math::IBaseFunctionMultiDim *
     new genie::utils::gsl::d2XSec_dQ2dy_E(fModel,fInteraction);
 }
 //____________________________________________________________________________
+genie::utils::gsl::d2XSec_dlog10xdlog10Q2_E::d2XSec_dlog10xdlog10Q2_E(
+     const XSecAlgorithmI * m, const Interaction * i, double scale) :
+ROOT::Math::IBaseFunctionMultiDim(),
+fModel(m),
+fInteraction(i),
+fScale(scale)
+{
+  
+}
+genie::utils::gsl::d2XSec_dlog10xdlog10Q2_E::~d2XSec_dlog10xdlog10Q2_E()
+{
+
+}
+unsigned int genie::utils::gsl::d2XSec_dlog10xdlog10Q2_E::NDim(void) const
+{
+  return 2;
+}
+double genie::utils::gsl::d2XSec_dlog10xdlog10Q2_E::DoEval(const double * xin) const
+{
+// inputs:  
+//  log10(x) [-]
+// log10(Q2) [-]
+// outputs: 
+//   differential cross section [10^-38 cm^2]
+//
+  double  log10x = xin[0];
```

Comment:
I appreciate the clarity of the passage here, but this function is supposed to be called a lot of times, can we avoid memory duplication here? 
Just pass xin[0] and xin[1] at the KinePtr setters